src/dex/curve-v1/curve-v1.ts

```diff
@@ -681,41 +695,46 @@ export class CurveV1 extends SimpleExchange implements IDex<CurveV1Data> {
     blockNumber: number,
     // list of pool identifiers to use for pricing, if undefined use all pools
     limitPools?: string[],
-    // from: Token,
-    // to: Token,
-    // amounts: bigint[],
-    // side: SwapSide,
-    // routeID: number,
-    // usedPools: { [poolIdentifier: string]: number } | null,
-    // version: string,
-    // isAmountsLinear: boolean,
-    // excludedPools: string[],
-    // // Last two are useful for testing purposes
-    // useCache: boolean,
-    // useEvent: boolean,
+    transferFees: TransferFeeParams = {
+      srcFee: 0,
+      destFee: 0,
+      srcDexFee: 0,
+      destDexFee: 0,
+    },
   ): Promise<ExchangePrices<CurveV1Data> | null> {
     try {
-      if (side === SwapSide.BUY) {
+      const isSell = side === SwapSide.SELL;
+      if (!isSell) {
         return null;
       }
-      const from = this.dexHelper.config.wrapETH(_from);
-      const to = this.dexHelper.config.wrapETH(_to);
+
+      const _isTokenTransferFeeToBeExchanged =
+        isTokenTransferFeeToBeExchanged(transferFees);
 
       // We first filter out pools which were explicitly excluded and pools which are already used
       // then for the good pools we set the boolean to be true for used pools
       // and for each pool we take the address.
-      const goodPoolConfigs = this.getPoolConfigs(from, to).filter(p => {
+      const goodPoolConfigs = this.getPoolConfigs(_from, _to).filter(p => {
+        let keepPool = true;
         if (!limitPools) {
-          return true;
+          keepPool = true;
+        } else {
+          const id = this.getPoolIdentifier(p.name);
+          keepPool = limitPools!.includes(id);
+        }
+
+        if (_isTokenTransferFeeToBeExchanged) {
+          // Fee on transfers supported only when flag is specified
+          return keepPool && !!p.isFeeOnTransferSupported;
+        } else {
+          return keepPool;
```

Comment:
It doesn't make sense to have this `keepPool` variable, when you can simply `return false;` at the appropriate places and `return true;` at the end.  Please refactor.